pinot-controller/src/main/java/org/apache/pinot/controller/recommender/io/InputManager.java

```diff
@@ -0,0 +1,519 @@
+
+
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.NONE)
+public class InputManager {
+  private final Logger LOGGER = LoggerFactory.getLogger(InputManager.class);
+
+  /******************************Deserialized from input json*********************************/
+  // Basic input fields
+  public RulesToExecute _rulesToExecute = new RulesToExecute(); // dictates which rules to execute
+  public Schema _schema = new Schema();
+  public SchemaWithMetaData _schemaWithMetaData = new SchemaWithMetaData();
+
+  public String _queryType = SQL; // SQL or PQL
+  public long _qps = DEFAULT_QPS;
+  public Map<String, Double> _queryWeightMap = new HashMap<>(); // {"queryString":"queryWeight"}
+  public String _tableType = OFFLINE;
+  public long _numMessagesPerSec = DEFAULT_NUM_MSG_PER_SEC; // messages per sec for kafka to consume
+  public long _numRecordsPerPush = DEFAULT_NUM_RECORDS_PER_PUSH; // records per push for offline part of a table
+  public long _latencySLA = DEFAULT_LATENCY_SLA; // latency sla in ms
+  public int _numKafkaPartitions = DEFAULT_NUM_KAFKA_PARTITIONS;
+
+  // The parameters of rules
+  public PartitionRuleParams _partitionRuleParams = new PartitionRuleParams();
+  public InvertedSortedIndexJointRuleParams _invertedSortedIndexJointRuleParams =
+      new InvertedSortedIndexJointRuleParams();
+  public BloomFilterRuleParams _bloomFilterRuleParams = new BloomFilterRuleParams();
+  public NoDictionaryOnHeapDictionaryJointRuleParams _noDictionaryOnHeapDictionaryJointRuleParams =
+      new NoDictionaryOnHeapDictionaryJointRuleParams();
+  public FlagQueryRuleParams _flagQueryRuleParams = new FlagQueryRuleParams();
+
+  // For forward compatibility: 1. dev/sre to overwrite field(s) 2. incremental recommendation on existing/staging tables
+  public ConfigManager _overWrittenConfigs = new ConfigManager();
+
+  /******************************Ignored by deserializer****************************************/
+  public Map<String, ColumnMetaData> _metaDataMap = new HashMap<>(); // meta data per column, complement to schema
+  long _sizePerRecord = 0;
+  Map<String, FieldSpec.DataType> _colnameFieldTypeMap = new HashMap<>();
+  Set<String> _dimNames = null;
+  Set<String> _metricNames = null;
+  Set<String> _dateTimeNames = null;
+  Set<String> _dimNamesInveredSortedIndexApplicable = null;
+  Map<String, Integer> _colNameToIntMap = null;
+  String[] _intToColNameMap = null;
+  Map<FieldSpec.DataType, Integer> _dataTypeSizeMap = new HashMap<FieldSpec.DataType, Integer>() {{
+    put(FieldSpec.DataType.INT, DEFAULT_INT_SIZE);
+    put(FieldSpec.DataType.LONG, DEFAULT_LONG_SIZE);
+    put(FieldSpec.DataType.FLOAT, DEFAULT_FLOAT_SIZE);
+    put(FieldSpec.DataType.DOUBLE, DEFAULT_DOUBLE_SIZE);
+    put(FieldSpec.DataType.BYTES, DEFAULT_BYTE_SIZE);
+    put(FieldSpec.DataType.STRING, DEFAULT_CHAR_SIZE);
+    put(null, DEFAULT_NULL_SIZE);
+  }};
+
+  /**
+   * Process the dependencies incurred by overwritten configs.
```

Comment:
Please see my comment above w.r.t explaining the purpose and usage of overwritten configs.